### ISSUE TITLE
Build Docker image upon GitHub version tag

### ISSUE THREAD

**Issue by bdwilliamson** [benkeser/slapnap#40]
It'd be great to have this functionality. It seems like it should go in .travis.yml, but I'm not sure where. Another option is to configure automatic builds on DockerHub, but that seems less satisfying.

**Comment by benkeser** [comment]
Well, the image will definitely build on travis for a version that is tagged, because tags are just specific commits and all commits to master are built. It's a matter of tagging the release on docker hub.

Alternatively, we could create a `dev` branch that we check builds of on travis, but do not push to DockerHub. That way `:latest` container will always be a stable build from `master`.

**Comment by bdwilliamson** [comment]
Hmm, then something's afoot, because the latest build on DockerHub is from 22 days ago and I tagged a release on GitHub 14 days ago. 

In the first workflow, would you have to manually tag releases on DockerHub to have them harmonized? If we can set up a way to auto-tag releases on DockerHub (that are also tagged on GitHub), that would be great.

I like the second workflow with `dev`, but ideally we would also have the auto-tagging feature.

**Comment by benkeser** [comment]
It may only build with PRs that I merge? I’m not sure actually. Don’t understand precisely how Travis does the verification of the encrypted credentials.

**Comment by bdwilliamson** [comment]
Reviving this issue because the PR you merged yesterday didn't trigger a Travis build; downstream effects are no update to the book and no update to the Docker image.

Unclear if this is related, but something (I think the correlates repo) ate up all of my Travis credits, so it's possible that this will never run while correlates builds are happening? Perhaps we should switch to GitHub Actions (which I've seen other people do recently)?

**Comment by benkeser** [comment]
Yeah it may be the case that I've used all my travis credits. Could you run (for now) the build scripts locally and push to docker hub, `gh-pages`, etc?

**Comment by bdwilliamson** [comment]
Sure, that seems like a good option. I'll leave the issue open in case we want to come back to this later.

**Comment by bdwilliamson** [comment]
I can update the book but I can't push to Docker Hub (I don't have the password for slapnap). Not sure if there is a secure way to share the password, but that's probably the easiest option (unless you want to build locally and push to Docker Hub yourself).
